group-coordinator/src/main/java/org/apache/kafka/coordinator/group/streams/StreamsGroup.java

```diff
@@ -197,6 +197,13 @@ public static class DeadlineAndEpoch {
      */
     private DeadlineAndEpoch metadataRefreshDeadline = DeadlineAndEpoch.EMPTY;
 
+    /**
+     * A flag to indicate whether a shutdown has been requested for this group.
+     * This has no direct effect inside the group coordinator, but is propagated to all members of the group.
+     * This is not persisted in the log.
```

Comment:
to understand this flag:
- we only clear when a group goes empty, this is when we assume all members acked
- since this is only in soft state, we have no guarantee that all members will eventually shutdown. it is a best effort nudge.

is this correct?

Reply:
Yes, I also wasn't sure about these things, so we discussed it in this thread: https://confluent.slack.com/archives/C06UA1XN326/p1743686382861189

The aim of the PR is to keep it simple, so:
 - The flag is best-effort, as in the old protocol
 - The flag remains set until either there is a coordinator failover, or all members have shutdown. The application is not intended to be restarted after this flag.

Reply:
I extended the comment a bit.